File: backend-common/src/main/java/cc/jktu/apps/common/util/BcryptUtil.java
package cc.jktu.apps.common.util;

import org.mindrot.jbcrypt.BCrypt;

public final class BcryptUtil {

    private BcryptUtil() {
    }

    public static String hashPassword(String plain) {
        return BCrypt.hashpw(plain, BCrypt.gensalt());
    }

    public static Boolean checkPassword(String plain, String hashed) {
        return BCrypt.checkpw(plain, hashed);
    }

}

File: backend-common/src/main/kotlin/cc/jktu/apps/common/util/BcryptUtil.kt
package cc.jktu.apps.common.util

import org.mindrot.jbcrypt.BCrypt

object BcryptUtil {

    fun hashPassword(plain: String): String {
        return BCrypt.hashpw(plain, BCrypt.gensalt())
    }

    fun checkPassword(plain: String, hashed: String): Boolean {
        return BCrypt.checkpw(plain, hashed)
    }
}